eng/pipelines/templates/jobs/smoke.tests.yml

```diff
@@ -179,51 +179,37 @@ jobs:
               return $line
             }
 
-            $dependencies | Out-File $env:REQUIREMENTS
+            $dependencies | Out-File $(requirements)
 
           displayName: Override requirements with pipeline build artifact versions
-          env:
-            REQUIREMENTS: $(requirements)
 
         - pwsh: |
-            $timeout = New-TimeSpan -Seconds 600
-            $stopwatch = [System.Diagnostics.Stopwatch]::StartNew()
             do {
               $failure = $false
-              try {
-                $ErrorActionPreference = "Stop"
-                pip --retries 10 install -r "$env:REQUIREMENTS" --no-deps --upgrade --no-cache-dir 2>&1
-              } catch {
+              pip --retries 10 install -r "$(requirements)" --no-deps --upgrade --no-cache-dir 2>&1
```

Comment:
Do we need to redirect stderr here? For most cases we generally just depend on the LASTEXITCODE like you have below.